src/FSharp.Charting.fs

```diff
@@ -1941,45 +1943,42 @@ namespace FSharp.Charting
-                                    | Charting.Docking.Left -> (max l DefaultExtraMarginForTitleIfPresent, t, r, b) 
-                                    | Charting.Docking.Right -> (l, t, max r DefaultExtraMarginForTitleIfPresent, b)
-                                    | Charting.Docking.Top -> (l, max t DefaultExtraMarginForTitleIfPresent, r, b) 
-                                    | Charting.Docking.Bottom -> (l, t, r, max b DefaultExtraMarginForTitleIfPresent)
-                                    | _ -> margins
-                                else
-                                    margins
+                let rec computeExtraDefaultMargins margins (srcSubChart:GenericChart) =
+                    if srcSubChart.Margin <> DefaultMarginForEachChart then
                         margins
+                    else
+                        match srcSubChart with
+                        | :? SubplotChart as subplot ->
+                            (margins, subplot.Charts) ||> List.fold computeExtraDefaultMargins
+                        | _ ->
+                            let legendAndTitleSubCharts = srcSubChart  :: (match srcSubChart with :? CombinedChart as cch -> cch.Charts | _ -> [])
+                            let getSingleSideMargins docking margin =
+                                match docking with
+                                | Charting.Docking.Left -> (margin, 0, 0, 0)
+                                | Charting.Docking.Right -> (0, 0, margin, 0)
+                                | Charting.Docking.Top -> (0, margin, 0, 0)
+                                | Charting.Docking.Bottom -> (0, 0, 0, margin)
+                                | _ -> (0, 0, 0, 0)
+
+                            let extraMarginsForLegend =
+                                match legendAndTitleSubCharts |> List.tryPick (fun ch -> ch.TryLegend) with
+                                | None -> (0, 0, 0, 0)
+                                | Some leg when leg.Enabled && not leg.IsDockedInsideChartArea ->
+                                    getSingleSideMargins leg.Docking DefaultExtraMarginForLegendIfPresent
+                                | _ -> (0, 0, 0, 0)
+
+                            let extraMarginsForTitle =
+                                match legendAndTitleSubCharts |> List.tryPick (fun ch -> ch.TryTitle) with
+                                | None -> (0, 0, 0, 0)
+                                | Some title when not title.IsDockedInsideChartArea ->
+                                    getSingleSideMargins title.Docking DefaultExtraMarginForTitleIfPresent
+                                | _ -> (0, 0, 0, 0)
+
+                            let pickMaxMargins (al, at, ar, ab) (bl, bt, br, bb) =
+                                (max al bl, max at bt, max ar br, max ab bb)
+
+                            [extraMarginsForLegend; extraMarginsForTitle]
+                            |> List.fold pickMaxMargins margins
```

Comment:
I pick the maximum margin for each side because that's what the old code did but I wonder if extra margins for each side shouldn't be added together instead.